src/h5web/visualizations/nexus/utils.ts

```diff
@@ -67,31 +62,35 @@ export function getNxDataGroup(
     return undefined;
   }
 
-  if (isNxDataGroup(entity)) {
-    return entity;
+  if (getAttributeValue(entity, 'NX_class') === 'NXdata') {
+    return entity; // `NXdata` group found
   }
 
   const defaultPath = getAttributeValue(entity, 'default');
-
-  if (typeof defaultPath !== 'string') {
-    return undefined;
+  if (defaultPath === undefined) {
+    return undefined; // group has no `default` attribute
   }
 
-  const relativePath = defaultPath.startsWith('/')
-    ? defaultPath.slice(1)
-    : defaultPath;
+  assertStr(defaultPath, `Expected 'default' attribute to be a string`);
+
+  const isAbsolutePath = defaultPath.startsWith('/');
+  const pathSegments = defaultPath.slice(isAbsolutePath ? 1 : 0).split('/');
 
-  const defaultEntity = relativePath.split('/').reduce<HDF5Entity | undefined>(
-    (previousEntity, currentComponent) => {
-      if (!previousEntity || !isGroup(previousEntity)) {
-        return undefined;
-      }
-      return getLinkedEntity(currentComponent, previousEntity, metadata);
+  const defaultEntity = pathSegments.reduce<HDF5Entity | undefined>(
+    (parentEntity, currSegment) => {
+      return parentEntity && isGroup(parentEntity)
+        ? getLinkedEntity(currSegment, parentEntity, metadata)
+        : undefined;
     },
-    defaultPath.startsWith('/') ? metadata.groups[metadata.root] : entity
+    isAbsolutePath ? metadata.groups[metadata.root] : entity
   );
```

Comment:
I had trouble going through this algorithm again, so I renamed and refactored a few things.